engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/sql/SimpleSQLReportDataFactory.java

```diff
@@ -130,9 +136,10 @@ protected synchronized Connection getConnection( final DataRow dataRow ) throws
     return connection;
   }
 
-  private int getBestResultSetType( final DataRow dataRow ) throws SQLException {
-    if ( "simple".equalsIgnoreCase( getConfiguration().getConfigProperty( //$NON-NLS-1$
-        ResultSetTableModelFactory.RESULTSET_FACTORY_MODE ) ) ) { //$NON-NLS-1$
+  @VisibleForTesting
```

Comment:
Is this Guava thing really necessary? This is a protected method that is only referenced from the same package (as the tests are local), so this should not be needed. Putting a new dependency in just for the sake of test annotations seems so wrong on so many levels. 

To Be honest, that method could even be public static, as the "globalConfig" variable you need from the local instance is actually fetched from a static context anyway. If you inline this, that becomes clear. And for static helper methods public does not break encapsulation so that is fine by all standards I'm aware of.